devito/finite_differences/finite_difference.py

```diff
@@ -161,184 +163,58 @@ def second_derivative(expr, **kwargs):
  f(x + h_x, y)*g(x + h_x, y)/h_x**2
     """
 
-    order = kwargs.get('order', 2)
-    dim = kwargs.get('dim')
-    diff = kwargs.get('diff', dim.spacing)
-
-    ind = [(dim + i * diff) for i in range(-int(order / 2),
-                                           int(order / 2) + 1)]
-
-    coeffs = finite_diff_weights(2, ind, dim)[-1][-1]
-    deriv = 0
-    all_dims = tuple(set((dim, ) +
-                     tuple([i for i in expr.indices if i.root == dim])))
-    for i in range(0, len(ind)):
-            subs = dict([(d, ind[i].subs({dim: d})) for d in all_dims])
-            deriv += coeffs[i] * expr.subs(subs)
-    return deriv.evalf(_PRECISION)
+    return generic_derivative(expr, dim, fd_order, 2, stagger=None)
 
 
 @check_input
-def generic_derivative(expr, deriv_order, dim, fd_order, **kwargs):
+def cross_derivative(expr, dims, fd_order, deriv_order, stagger=(None, None)):
```

Comment:
stagger=None
?
(None, None) as default doesn't make much sense
then, in the body:
```
stagger = stagger or [None]*len(dims) 
```

Reply:
reminder